### ISSUE TITLE
processFrame returns error RuntimeError: memory access out of bounds

### ISSUE THREAD

**Issue by diruuu** [DeepARSDK/quickstart-web-js#12]
Hi,
I got this `RuntimeError: memory access out of bounds` only if I call `processFrame` in loop continuously. 

In my app, I read a live view image coming from my camera in JPEG format every 30ms and then I use `processFrame` on deepAR to load the image into my canvas. It works fine for maybe 30 seconds. I can see the memory on my Task Manager got increased dramatically from time to time after I start the app and after that DeepAR just returns the `memory access out of bounds` error.

The issues seems coming from the wasm file. So my question, is there any some kind of garbage collection mechanism on the file? I'm not really familiar with web assembly but I can only guess there's a memory leak in your `wasm` file.

Let me know if there's any solution for this. We need to release the app soon.

**Comment by diruuu** [comment]
Found a solution for this.

I ended up modifying the `deepar.js` file to free up the memory.

```js
      e.processFrameRawPixels = function (q, B, C, A) {
        var K = e._malloc(q.length);
        e.HEAPU8.set(q, K);
        q = new ImageData(new Uint8ClampedArray(q), B, C);
        v.preProcessWithBuffer(q);
        Promise.all(
          aa
            .filter(function (P) {
              return P.Jh();
            })
            .map(function (P) {
              return P.Ph().then(P.Nh);
            })
        )
          .then(function () {
            v && v.postProcess();
          })
          .catch(function (P) {
            console.log(P);
          })
          .finally(function () {
            e._processFrameRawPixels(K, B, C, A);
            w || e._render();
            e._free(K); // <----------------- I ADDED THIS
          });
      };
```

You might consider adding this to the SDK file because the memory leak is really annoying at the moment.

**Comment by MateGasparini** [comment]
Hi @diruuu!
Thanks for reporting this.
The issue has been fixed in the newest Web SDK (v4.0.0).
